src_python/fl_client.py

```diff
@@ -0,0 +1,223 @@
+import socket
+import pickle
+import select
+import sys
+import numpy as np
+import pandas as pd
+import logging
+from timeit import default_timer as timer
+import time
+
+arg_names = [
+    'path_weights',
+    'path_nodes',
+    'path_edges',
+    'graph_id',
+    'partition_id',
+    'epochs',
+    'IP',
+    'PORT'
+]
+
+args = dict(zip(arg_names, sys.argv[1:]))
+
+partition_id = args['partition_id']
+
+logging.basicConfig(
+    level=logging.INFO, 
+    format='%(asctime)s : [%(levelname)s]  %(message)s',
+    handlers=[
+        logging.FileHandler(f'client_{partition_id}.log'),
+        logging.StreamHandler(sys.stdout)
+    ]
+)
+class Client:
+
+    def __init__(self, MODEL, graph_params, weights_path, graph_id, partition_id, epochs = 10, IP = socket.gethostname(), PORT = 5000, HEADER_LENGTH = 10):
+
+        self.HEADER_LENGTH =  HEADER_LENGTH
+        self.IP = IP
+        self.PORT = PORT
+
+        self.weights_path = weights_path
+        self.graph_id = graph_id
+        self.partition_id = partition_id
+        self.epochs = epochs
+
+        self.graph_params = graph_params
+
+        self.client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        connected = False
+        while not connected:
+            try:
+                self.client_socket.connect((IP, PORT))
+            except ConnectionRefusedError:
+                time.sleep(5)
+            else:
+                logging.info('Connected to the server')
+                connected = True
+        
+
+        self.MODEL = MODEL
+        self.STOP_FLAG = False
+        self.rounds = 0
+
+
+    def send_model(self):
+
+        # svae model weights
```

Comment:
Change `# svae model weights` to `# save model weights`

Reply:
done

Reply:
Ok